app/actions/local/channel.ts

```diff
@@ -279,6 +279,53 @@ export const storeMyChannelsForTeam = async (serverUrl: string, teamId: string,
     return {models: flattenedModels};
 };
 
+export const updateMyChannelFromWebsocket = async (serverUrl: string, channelMember: ChannelMembership, prepareRecordsOnly = false) => {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const member = await getMyChannel(operator.database, channelMember.channel_id);
+    if (member) {
+        member.prepareUpdate((m) => {
+            m.roles = channelMember.roles;
+        });
+        if (!prepareRecordsOnly) {
+            operator.batchRecords([member]);
+        }
+    }
+    return {model: member};
+};
+
+export const updateChannelInfoFromChannel = async (serverUrl: string, channel: Channel, prepareRecordsOnly = false) => {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const storedInfo = await getChannelInfo(operator.database, channel.id);
+    if (storedInfo) {
+        storedInfo.prepareUpdate((i) => {
+            i.purpose = channel.purpose;
+            i.header = channel.header;
+        });
+        return {model: storedInfo};
+    }
+    const newInfo = (await operator.handleChannelInfo({channelInfos: [{
+        guest_count: 0,
+        member_count: 0,
+        pinned_post_count: 0,
```

Comment:
I made all of them optional, since there are two "sources":
- I have the stats, so I have the guest_count, member_count and pinne_post_count but no header nor purpose
- I have the header and purpose, but not the other fields.